### PR TITLE
chore: Adds internal badge display feature to button dropdown

### PR DESCRIPTION
**Pull request by katiegeorge**
### Description

This PR adds an internal `badge` property to the Button Dropdown components. This adds a red circle to the top right corner of icons in the items, as well as the trigger.

This also changes the badge color from blue to red. We discussed this as a design team and decided we should not have two different colors that indicate notifications, and red is generally the industry standard (and does not signal 'negative').

This is a pre-cursor to drawers, which will use this to determine if overflow menu items have badges or not.

Screenshots:
image

Related links, issue #, if available: n/a

### How has this been tested?

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pages/button/permutations.page.tsx [hunk 1]
```diff
@@ -66,6 +69,7 @@ const permutations = createPermutations<ButtonProps>([
     children: ['With custom svg icon'],
     ariaLabel: ['With custom svg icon'],
     variant: ['normal', 'icon', 'inline-icon', 'primary'],
+    badge: [true, false],
```
Comment (just-boris) [inline pages/button/permutations.page.tsx:72]: Adding badges to existing set multiplied all existing permutations twice and exceeded the page height limit for screenshot tests. There is a warning on this page:

image


We need to create a separate set specifically for badge use-cases below

File: src/button-dropdown/__tests__/button-dropdown-items.test.tsx [hunk 2]
```diff
@@ -419,6 +420,14 @@ const items: ButtonDropdownProps.Items = [
         wrapper.openDropdown();
         expect(wrapper.findItemById('i1')!.findAllByClassName(IconWrapper.rootSelector)).toHaveLength(2);
       });
+
+      it('should render badge when defined', () => {
```
Comment (just-boris) [inline src/button-dropdown/__tests__/button-dropdown-items.test.tsx:424]: Should we add two more tests for the trigger badge rendering 1) badge is rendered when there are items with badges. 2) badge is not rendered if variant is not `icon`

**Comment by just-boris** [comment]
Renamed the PR title, because this is what is going to be displayed in release notes

File: src/button-dropdown/__tests__/button-dropdown-items.test.tsx (hunk 2 above)
Comment (katiegeorge) [reply src/button-dropdown/__tests__/button-dropdown-items.test.tsx]: Added (1) but not (2) since that isn't relevant anymore if this is internal-only

**Review by just-boris — DISMISSED** [review]

**Review by just-boris — APPROVED** [review]

**Pull request merged**